fix(mount, master): Fix path by inode permissions

rolysr opened the pull request:
## Fix path by inode improvements
1) Fix behavior when the given inode was not existing.
2) Ensure that the user has permissions to all nodes in the path that contains the given inode.

dmga44 reviewed (approved):
LGTM

wg1k reviewed (commented):
How are you testing your changes?

antuan96314 reviewed (commented):
Consider adding/modifying a test to verify new behavior.

rolysr committed:
fix(mount): Fix path by inode when not existing

Previous path by inode hidden file was having an inconsistent
behavior when the requested inode was not existing. This commit
fixes that scenario.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

rolysr committed:
fix(mount): Check permissions in path by inode

Previous implementation of path by inode feature was not checking
if a given user requesting the corresponding data was having
access to each of the nodes in the path for the given inode.
This commit fixes that issue.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

rolysr commented:
@antuan96314 @ictus4u please check latest changes.
